### PR TITLE
Fixes #747: convert sample github_hmac to use classic Go template

### LINKED ISSUE #747: Convert sample github_hmac to use classic Go template
Convert sample github_hmac to use classic Go template

https://github.com/openfaas/faas-cli/tree/master/sample/github_hmac

This will be less to maintain over time, a Dockerfile isn't required for the sample.

### PR DESCRIPTION
**Pull request by Amalkh5**
## Description
This change convert sample github_hmac to use classic Go template

## Motivation and Context

#747 
- [x] I have raised an issue to propose this change ([required](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md))

## How Has This Been Tested?
yes 
```
$ echo ddddd | go run main.go                                                               
Expected: cf23df2207d99a749a74fbe169a035e633b65d94
MessageMAC: cf23df2207d99a74fbe169e3eba035e633b65d94
CalculatedMAC: a823133e3f0aae25c6195f97c6f1d67ac559b94e
The message was not from your GitHub account, or you don't share the same secret.
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by alexellis** [comment]
Thanks for sending your first PR. Please could you spend some time to read the contribution guide?

**Comment by Amalkh5** [comment]
@alexellis I do not understand because my commit is already signoff. 
please check https://github.com/Amalkh5/faas-cli/commit/4eac5058c8cbc26930c73a8307fc91f8586cd180

**Comment by alexellis** [comment]
You may need confusing signoff with cryptographic GPG signing. They are very different and both are explained in the contribution guide which is why I suggested that you may benefit from reading it.

Alternatively you can run the command given to you by Derek.

**Comment by Amalkh5** [comment]
@alexellis fixed

**Comment by alexellis** [comment]
```
$ echo ddddd | go run main.go                                                               
```

^ this is not evidence of testing, the task is to delete main.go and to use the `go` template, which means having handler.go.

```
export FN="INSERT_NAME_HERE"
faas-cli new --lang go $FN
```
Then in the $FN/handler.go implement the logic

Finally you need to actually test the function and deploy it. For help on using OpenFaaS check the workshop - the first few labs explain how to build and deploy - https://github.com/openfaas/workshop/

**Comment by Amalkh5** [comment]
@alexellis yes I totally misunderstood this but I fixed it and there is the test

1. create new function 
```
$ export FN="githubhmac"    
$ faas-cli new --lang go $FN

Folder: githubhmac created.
  ___                   _____           ____
 / _ \ _ __   ___ _ __ |  ___|_ _  __ _/ ___|
| | | | '_ \ / _ \ '_ \| |_ / _` |/ _` \___ \
| |_| | |_) |  __/ | | |  _| (_| | (_| |___) |
 \___/| .__/ \___|_| |_|_|  \__,_|\__,_|____/
      |_|


Function created in folder: githubhmac
Stack file written: githubhmac.yml

Notes:
You have created a new function which uses Golang 1.11 and the Classic
OpenFaaS template.

To include third-party dependencies, use a vendoring tool like dep:
dep documentation: https://github.com/golang/dep#installation

For high-throughput applications, we recommend using the golang-http
or golang-middleware templates instead available via the store.
```

2. building the image 
```
 $ faas-cli build -f githubhmac.yml 

[0] > Building githubhmac.
Clearing temporary build folder: ./build/githubhmac/
Preparing ./githubhmac/ ./build/githubhmac//function
Building: githubhmac:latest with go template. Please wait..
Sending build context to Docker daemon  8.704kB
Step 1/30 : FROM openfaas/classic-watchdog:0.18.1 as watchdog
 ---> 94b5e0bef891
Step 2/30 : FROM golang:1.13-alpine3.11 as builder
 ---> e1fd9820be16
Step 3/30 : ARG ADDITIONAL_PACKAGE
 ---> Using cache
 ---> 9617d0823a67
Step 4/30 : ARG CGO_ENABLED=0
 ---> Using cache
 ---> bfca8715b084
Step 5/30 : ARG GO111MODULE="off"
 ---> Using cache
 ---> b19274d6603b
Step 6/30 : ARG GOPROXY=""
 ---> Using cache
 ---> 0686f28d8947
Step 7/30 : ARG GOFLAGS=""
 ---> Using cache
 ---> 06147c95ca8f
Step 8/30 : COPY --from=watchdog /fwatchdog /usr/bin/fwatchdog
 ---> Using cache
 ---> 9678e445ce73
Step 9/30 : RUN chmod +x /usr/bin/fwatchdog
 ---> Using cache
 ---> 349f42bdd8bd
Step 10/30 : ENV CGO_ENABLED=0
 ---> Using cache
 ---> 4f22b060a531
Step 11/30 : WORKDIR /go/src/handler
 ---> Using cache
 ---> f86970efa175
Step 12/30 : COPY . .
 ---> 954c60208870
Step 13/30 : RUN cat function/GO_REPLACE.txt >> ./go.mod || exit 0
 ---> Running in 80f76d265f8b
cat: can't open 'function/GO_REPLACE.txt': No such file or directory
Removing intermediate container 80f76d265f8b
 ---> 59041edde19e
Step 14/30 : RUN test -z "$(gofmt -l $(find . -type f -name '*.go' -not -path "./vendor/*" -not -path "./function/vendor/*"))" || { echo "Run \"gofmt -s -w\" on your Golang code"; exit 1; }
 ---> Running in d103a034ce4c
Removing intermediate container d103a034ce4c
 ---> 910813e1b53e
Step 15/30 : WORKDIR /go/src/handler/function
 ---> Running in 599b295b38a6
Removing intermediate container 599b295b38a6
 ---> 2b0708a0df5a
Step 16/30 : RUN go test ./... -cover
 ---> Running in 83ecf4cbf3e6
?       handler/function        [no test files]
Removing intermediate container 83ecf4cbf3e6
 ---> d5dfd3b643a9
Step 17/30 : WORKDIR /go/src/handler
 ---> Running in d8a12cea27c1
Removing intermediate container d8a12cea27c1
 ---> 48fe7ead08da
Step 18/30 : RUN CGO_ENABLED=${CGO_ENABLED} GOOS=linux     go build --ldflags "-s -w" -a -installsuffix cgo -o handler .
 ---> Running in 9f7dfa11c74c
Removing intermediate container 9f7dfa11c74c
 ---> 35bf88a30e5e
Step 19/30 : FROM alpine:3.11
 ---> e7d92cdc71fe
Step 20/30 : RUN apk --no-cache add ca-certificates     && addgroup -S app && adduser -S -g app app     && mkdir -p /home/app     && chown app /home/app
 ---> Using cache
 ---> fb274ad8deb8
Step 21/30 : WORKDIR /home/app
 ---> Using cache
 ---> 44c32d6a9077
Step 22/30 : COPY --from=builder /usr/bin/fwatchdog         .
 ---> Using cache
 ---> 44974e0c6a3c
Step 23/30 : COPY --from=builder /go/src/handler/function/  .
 ---> 5fcf9f954e60
Step 24/30 : COPY --from=builder /go/src/handler/handler    .
 ---> 3380248dd0d0
Step 25/30 : RUN chown -R app /home/app
 ---> Running in 247df6c059dc
Removing intermediate container 247df6c059dc
 ---> b47f7e5c5b24
Step 26/30 : USER app
 ---> Running in d585a83412df
Removing intermediate container d585a83412df
 ---> 19b86319f55b
Step 27/30 : ENV fprocess="./handler"
 ---> Running in 434f78dafdcd
Removing intermediate container 434f78dafdcd
 ---> c7ce9ff36ce4
Step 28/30 : EXPOSE 8080
 ---> Running in 42cae972f446
Removing intermediate container 42cae972f446
 ---> 06b0cc2e825b
Step 29/30 : HEALTHCHECK --interval=3s CMD [ -e /tmp/.lock ] || exit 1
 ---> Running in 381f2fe54cbc
Removing intermediate container 381f2fe54cbc
 ---> 43b59dba4ab3
Step 30/30 : CMD ["./fwatchdog"]
 ---> Running in f40453dee4ec
Removing intermediate container f40453dee4ec
 ---> 84d3659dcba9
Successfully built 84d3659dcba9
Successfully tagged githubhmac:latest
Image: githubhmac:latest built.
[0] < Building githubhmac done in 13.32s.
[0] Worker done.

Total build time: 13.32
```

3. deploy the function 
```
$ faas-cli deploy -e secret_key=hello -e Http_X-Hub-Signature="sha1=cf23df2207d99a749a74fbe169a035e633b65d94" -f githubhmac.yml

Deploying: githubhmac.
WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.

Deployed. 202 Accepted.
URL: http://127.0.0.1:8080/function/githubhmac
```

4. test the function

```
curl -X POST \
  http://127.0.0.1:8080/function/githubhmac \
  -d ddddd
Expected: cf23df2207d99a749a74fbe169a035e633b65d94
MessageMAC: cf23df2207d99a749a74fbe169a035e633b65d94
CalculatedMAC: b51f4760689148e80586427060c98763ed865cc4
The message was not from your GitHub account, or you don't share the same secret
```

Thanks for clarifying the issue.

File: sample/githubhmac/handler.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
+package function
```
Comment (alexellis) [inline sample/githubhmac/handler.go]: Why do we have `github-hmac` and `githubhmac` as folders now?

File: sample/github-hmac/handler.go [hunk 2]
```diff
@@ -1,33 +1,26 @@
...
 )
 
-func main() {
-	input, err := ioutil.ReadAll(os.Stdin)
-	if err != nil {
-		log.Fatalf("Unable to read standard input: %s", err.Error())
-	}
-
+// Handle a serverless request
+func Handle(req []byte) string {
```
Comment (alexellis) [inline sample/github-hmac/handler.go]: This file is repeated twice in the PR?

**Comment by alexellis** [comment]
Please create a stack.yml in the `faas-cli/sample/` folder that contains the entry for the HMAC function

**Comment by alexellis** [comment]
@Amalkh5 why did you close this PR?

**Comment by Amalkh5** [comment]
@alexellis I closed this PR #779 due to some issues with the branch, I opened a new one where I fixed everything you ask for.

**Pull request closed without merging**